File: JavaParser/Boolean_Expression.java

public class Boolean_Expression implements Expression

/**********************************************
 * expression is modeling an arithmetic expression
 */
{
	private Relational_Operator ropt;
	
	private Expression expr1;
	
	private Expression expr2;
	
	public Boolean_Expression()
	{	}
	
	public Boolean_Expression(Relational_Operator ropt, Expression expr1, Expression expr2)
	{
		if (ropt == null)
			throw new IllegalArgumentException("ropt error /BooleanExpression");
		if (expr1 == null)
			throw new IllegalArgumentException("expr1 error /BooleanExpression");
		if (expr2 == null)
			throw new IllegalArgumentException("expr2 error /BooleanExpression");
		
		this.ropt = ropt;
		this.expr1 = expr1;
		this.expr2 = expr2;
	}
	
	@Override
	public int evaluate()
	
	/*************************************
	 * should be returning a boolean
	 */
	{
		
		// if expression (whatever relational operator) expression 2, then true = 1 else false = 0
		int value = 0;
		if (ropt == Relational_Operator.gt_opt)
			value = expr1.evaluate() > expr2.evaluate() ? 1 : 0;
		else if (ropt == Relational_Operator.ge_opt)
			value = expr1.evaluate() >= expr2.evaluate() ? 1 : 0;
		else if (ropt == Relational_Operator.le_opt)
			value = expr1.evaluate() <= expr2.evaluate() ? 1 : 0;		
		else if (ropt == Relational_Operator.lt_opt)
			value = expr1.evaluate() < expr2.evaluate() ? 1 : 0;
		else if (ropt == Relational_Operator.eq_opt)
			value = expr1.evaluate() == expr2.evaluate() ? 1 : 0;
		else if (ropt == Relational_Operator.ne_opt)
			value = expr1.evaluate() != expr2.evaluate() ? 1 : 0;
		
		return value;
	}
}

File: PythonParser/BooleanExpression.py
__author__ = 'ravipatel'

from RelationalOperator import RelationalOperator

class BooleanExpression(object):

    def __init__(self, Relat_Opt, expr1, expr2):
        if (Relat_Opt is None):
            raise ValueError("Error in relational operator in the relational operator class")
        self.Relat_Opt = Relat_Opt
        if (expr1 is None):
            raise ValueError("Error in expr1 in the relational operator class")
        self.expr1 = expr1
        if (expr2 is None):
            raise ValueError("Error in expr2 in the relational operator class")
        self.expr2 = expr2

    def evaluate(self):
        if (self.Relat_Opt == RelationalOperator.le_opt):
            answer = (self.expr1.evaluate() <= self.expr2.evaluate())
        elif (self.Relat_Opt == RelationalOperator.ge_opt):
            answer = (self.expr1.evaluate() >= self.expr2.evaluate())
        elif (self.Relat_Opt == RelationalOperator.lt_opt):
            answer = (self.expr1.evaluate() < self.expr2.evaluate())
        elif (self.Relat_Opt == RelationalOperator.gt_opt):
            answer = (self.expr1.evaluate() > self.expr2.evaluate())
        elif (self.Relat_Opt == RelationalOperator.eq_opt):
            answer = (self.expr1.evaluate() == self.expr2.evaluate())
        elif (self.Relat_Opt == RelationalOperator.ne_opt):
            answer = (self.expr1.evaluate() != self.expr2.evaluate())
        else:
            raise ValueError("Error")
        return answer
